### PR TITLE
TimePicker : Ability to define dialOnly / inputOnly modes

### PR DESCRIPTION
**Pull request by LempereurBenjamin**
## Content
- Add dialOnly mode to the TimePicker
- Add inputOnly mode to the TimePicker

## Issues fixed by this PR
- https://github.com/flutter/flutter/issues/85013

## Comments
Inspired by this abandoned PR : https://github.com/flutter/flutter/pull/85189

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by LempereurBenjamin** [comment]
Dear @darrenaustin ,
Please can you review this PR ?
I mentioned you because you already reviewed the original PR (https://github.com/flutter/flutter/pull/85189).
Thank you.

File: packages/flutter/lib/src/material/time_picker.dart [hunk 1]
```diff
@@ -70,6 +70,12 @@ enum TimePickerEntryMode {
 
   /// Text input.
   input,
+
+  /// Only tapping/dragging on a clock dial.
+  dialOnly,
+
+  /// Only text input.
+  inputOnly
```
Comment (darrenaustin) [inline packages/flutter/lib/src/material/time_picker.dart:78]: Can we expand on the docs for all four of of these values to be consistent with how they are described for the equivalent in the Date Picker: 

https://github.com/flutter/flutter/blob/cb9a1d623ccd460c3744ed019af76496c47a353e/packages/flutter/lib/src/material/date.dart#L158

**Review by darrenaustin — CHANGES_REQUESTED** [review]
Thanks for the contribution. Looks great. Can we just update the docs for `TimePickerEntryMode` to more closely match the `DatePickerEntryMode`? I think that would make it a little clearer what the different values are used for.

File: packages/flutter/lib/src/material/time_picker.dart (hunk 1 above)
Comment (LempereurBenjamin) [reply packages/flutter/lib/src/material/time_picker.dart:86]: Hello @darrenaustin,
Thank you for the feedback. I made the change in the TimePickerEntryMode doc to match the DatePicker equivalent.
Benjamin.

**Review by darrenaustin — APPROVED** [review]
Sorry for the delay in response. Thanks for the contribution.

LGTM

**Review by Piinks — APPROVED** [review]
Flutter_LGTM

**Review by justinmc — APPROVED** [review]
LGTM 👍

**Comment by Vedsaga** [comment]
@darrenaustin  which flutter version this is supported? ca you pls confirm

**Comment by justinmc** [comment]
This has been released on all channels including stable.  I think it went in at v.3.3.0-0.0.pre.  Definitely the latest stable v3.3.5 has it.

**Pull request merged**